zulipterminal/core.py

```diff
@@ -109,6 +118,15 @@ def __init__(
         # Register new ^C handler
         signal.signal(signal.SIGINT, self.exit_handler)
 
+
+    @property
+    def uri(self) -> str:
+        return self._uri
+
+    def set_uri(self, value: str) -> None:
+        self._uri = value
+        self.uri_updated_event.set()
```

Comment:
Instead of simply setting the URI, is it possible for you to add the link generated for the uploaded file and append it to the `write_box`? Maybe that can eliminate the need to wait for the upload API to provide response.